### DISCUSSION TITLE
Global "Version"?

### DISCUSSION THREAD

**Discussion by richb-hanover** [sqm-autorate/sqm-autorate#146] [Operational Development]
In https://github.com/sqm-autorate/sqm-autorate/issues/139#issuecomment-1036394686, I suggested that it would be helpful to update the version number in the `sqm-autorate.lua` file so that we can tell the difference between different test branches.

Perhaps something "stronger" is warranted - a global "Version" that could be incorporated within the log messages of `sqm-autorate.lua` and possibly reported during the installation/configuration process. 

Thanks.

**Comment by Lochnair** [comment]
As mentioned in another comment, I think it would be neat to have the setup script handle setting the commit ID (and branch if applicable) automatically, so we have the exact commit when debugging something.

Not sure what the most sane way to set the values so script can find them, but we could always have some simple placeholders in the main script that the setup script can replace, e.g.:
```lua
-- The versioning value for this script
local _VERSION = "0.5.0"
local _BRANCH = "|BRANCH|"
local _COMMIT = "|COMMIT|"
```

Then do a quick sed replace in the setup script.

Had a quick look, and you can get the latest commit in a branch by doing `curl https://api.github.com/repos/sqm-autorate/sqm-autorate/branches/develop/main`:

```json
{
  "name": "develop/main",
  "commit": {
    "sha": "137c0dd62368f2feac335712a4f67e4cccaf44ae",
    "node_id": "C_kwDOGgi9g9oAKDEzN2MwZGQ2MjM2OGYyZmVhYzMzNTcxMmE0ZjY3ZTRjY2NhZjQ0YWU",
    "commit": {
      "author": {
        "name": "Jack",
        "email": "46714706+jeverley@users.noreply.github.com",
        "date": "2022-02-10T18:01:52Z"
      },

...
```

So definitely doable should we want to do it this way.

**Comment by richb-hanover** [comment]
Something like this would be enormously useful. But as my former business partner used to say, "One insurmountable hurdle at a time" :-)

**Reply by chas-iot** [reply]
I'll pick this up in the setup PR

**Comment by chas-iot** [comment]
This enhancement is merged into `develop/main` and will be in the next release
